models/yolo_people_fire_smoke/main.py

```
@@ -0,0 +1,109 @@
+from ultralytics import YOLO
+import cv2, time
+from collections import deque
+from hud import draw_hud, draw_boxes
+
+people_model = YOLO("yolov8n.pt") # People detection
+fire_model = YOLO("best.pt") # Fire and smoke detection
+
+colors = {
+    'person': (255, 0, 0), # Blue
+    'fire': (255, 0, 255), # Purple
+    'smoke': (0, 255, 255) # Yellow
+}
+
+fps_hist = deque(maxlen=30)
+inf_hist = deque(maxlen=30)
+drop_hist = deque(maxlen=30) # Stores average dropped frames per second
+t_prev = time.time()
+
+source_video = "original_short_demo.mp4"
```

If `source_video` is invalid or missing, `cap` may silently fail until `ret` becomes `False`. Consider checking `cap.isOpened()` immediately after initialization and logging an explicit error. This will make debugging incorrect paths or missing files easier.